assets/src/application/ui/display/EntityListLegend/index.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Row, Col } from 'antd';
+
+import { EspressoIcon } from '@appDisplay/espressoIcon';
+import { LegendConfig } from './types';
+import './style.scss';
+
+interface Props {
+	legendConfig: LegendConfig;
+}
+
+const EntityListLegend: React.FC<Props> = ({ legendConfig }) => {
+	const { icons, swatches } = legendConfig;
+
+	return (
+		<Row className='ee-row'>
+			<Col span={12}>
+				<dl className='ee-list-table-legend'>
```

Comment:
Although I really like the fact that you used a description list here, which is the most semantic markup for this use case, this isn't responsive the way it is setup.

I think we have two options here:

1. convert this completely to use AntD `Row` and `Col` components but use their responsive sizing props

ie:

```js
		<Row className='ee-row ee-list-table-legend' gutter={[16, { xs: 8, sm: 16, md: 24, lg: 32 }]}>
			{Object.entries(icons).map(([icon, description]) => {
				return (
					<Col xs={{ span: 24 }} sm={{ span: 12 }} md={{ span: 8 }} lg={{ span: 4 }}>
						<EspressoIcon icon={icon} svgSize={18} />
						<span className='ee-legend-description'>{description}</span>
					</Col>
				);
			})}
			{Object.entries(swatches).map(([swatchClassName, description]) => {
				return (
					<Col xs={{ span: 24 }} sm={{ span: 12 }} md={{ span: 8 }} lg={{ span: 4 }}>
						<span
							className={classNames('ee-status-legend', 'ee-status-background-color-' + swatchClassName)}
						></span>
						<span className='ee-legend-description'>{description}</span>
					</Col>
				);
			})}
		</Row>
```

2. drop the AntD components completely and use a description list (because I think your choice of dl, dt was great), but then use CSS grid to control the display

The former is easier but I think the latter would be better, at least from a semantics perspective which is somewhat important for a11y. We could also abstract either of those to be components in the app layer UI folder, whereas an abstract "List Table Legend" could go in with the Entity List component similar to how the filter bar and pagination are bundled with it. The specific "legend" components would of course need to go in the appropriate `/domain/` folder.

As well, I think this would be better if it simply mapped a single array of legend items to whatever kind of grid we end up using. 

How about we do the following:

- create a `LegendItem` component
- use additional components like `LegendItemIcon` and `LegendItemSwatch` for controlling those two aspects of things
- have those passed in via a single prop that is typed for those two components
- then either create:
     - `/application/ui/layout/grid` which wraps the AntD grid and sets up some common sizes for various breakpoints
     - `/application/ui/layout/descriptionList` which would use `<dl>`, `<dt>`, and `<dd>` for markup and CSS grid for controlling the display and responsiveness

This will better follow Single Responsibility Principle (SRP) imho since the individual legend item components will control their display as opposed to the list doing so. This allows for more variability of items to be added to these kinds of lists. For example, what if in the future we want to add a third kind of legend item? I also think having a list of one specific component will make it easier to handle adding items via the registry since it can be restricted to that specific type.

Reply:
@tn3rb please elaborate this:
`have those passed in via a single prop that is typed for those two components`

Reply:
in its simplest form, imagine the `LegendItem` component having two props:

- symbol: for displaying an icon or status swatch

- description: the text that explains what the legend item symbol represents

The typing for the props might then be something like :

```ts
interface LegendItemProps {
	symbol: Icon | Swatch; 
	description: string;
}
```

where `Icon | Swatch` would correspond to the `LegendItemIcon` and `LegendItemSwatch` components.

Now that said, there's multiple ways this could be done.

If you go the route of creating a more abstracted "DescriptionList" component, then instead of having  `LegendItem`, `LegendItemIcon`, and `LegendItemSwatch` components, it might be better to have generic `ListItem` and `ColorSwatch` components that we use in conjunction with `EspressoIcon` and then simply combine those together when creating an `AdminLegend` component.

The primary reason I'm suggesting this level of abstraction is because it's very likely that we will require these kinds of components for other use cases. As well, when it comes to extending these things via Manzoor's registry, we want to be able to easily restrict what gets added to things. So in this case, we don't want add-ons or third party devs co-opting an admin legend to add in some completely different and unexpected content. So only having one specific component that can be registered will clamp down on what can be done with those elements.